tensorboard/uploader/uploader_main.py

```diff
@@ -501,10 +502,17 @@ def _get_intent(flags):
 
 
 def _get_server_info(flags):
-  if flags.api_endpoint:
-    return server_info_lib.create_server_info(flags.origin, flags.api_endpoint)
-  else:
-    return server_info_lib.fetch_server_info(flags.origin)
+  if flags.api_endpoint and not flags.origin:
+    return server_info_lib.create_server_info(
+        "http://localhost:8080", flags.api_endpoint)
+  origin = flags.origin or _DEFAULT_ORIGIN
+  server_info = server_info_lib.fetch_server_info(
+      flags.origin or _DEFAULT_ORIGIN)
```

Comment:
this should just be `origin` here right?

Reply:
Yes; thanks.